internal/hcsoci/resources_lcow.go

```diff
@@ -21,6 +21,7 @@ import (
 
 const lcowMountPathPrefix = "/mounts/m%d"
 const lcowGlobalMountPrefix = "/run/mounts/m%d"
+var lcowGlobalMountCount = 0
```

Comment:
I think we should have this mount count be a field on the UVM.

Reply:
Fixed.

Reply:
I agree. Due to Danny's work we have been talking about longer term goals for this code that might involve moving more of it out of the UVM, but I think right now we just need to make this fix as robust as possible.

With that in mind, is there any reason the caller of `AddSCSI` should ever need to specify the UVM path? I think we should just put all of the UVM path determination logic in `scsi.go`, and return the path to use, but not accept one as a parameter. It's already really weird because we essentially accept a "suggested" path to use, but e.g. ignore it if the UVM is already moutned somewhere else.

Moving this into `scsi.go` also means we will guard incrementing the global count with the UVM's mutex.

Reply:
I don't think I like the idea of maintaining a count for the scsi path index. Say we have two containers, the first allocates /run/mounts/m1 and the second allocates /run/mounts/m2. When the first container exits and (given that it was the only one with a reference of m1) removes the m1 mount, the counter would not be able to reclaim this index. If we don't think this will ever cause a problem, then it's fine to keep it this way, but I think it would be better overall to use some other way to creating a unique identifier for the mounts.

Reply:
I agree with @katiewasnothere point about recycling counters. If the UVM is running for long enough we might end up using all 2^32 numbers (however, we will have to mount a LOT of disks for that to happen).

Reply:
I'm fine with it not being an integer index. I just think we should try to have SCSI tell you where it's mounted, instead of the other way around.

Reply:
I think finding the unique host path inside the UVM is not the job of SCSI at all. In fact the check for existing mounts (and returning that mount path) should also be done outside of SCSI even before we call AddSCSI. If we want to now add a PMEM device to the container then we will want to generate a unique mount paths for that PMEM device. If the PMEM device management code generates its own unique paths then those paths can collide with the paths generated for SCSI. I think unique path creation should be maintained at the UVM level.

Reply:
If we switch to using a guid for the unique identifier of the mount, I don't think path collision is much of a concern. Maybe finding if the device already exists on the uvm and then if not finding a unique path in the uvm to create a new scsi device can just be an additional function in `scsi.go` that could be called outside of that file if you want to keep it out of the AddSCSI function.

Reply:
GUIDs are unnecessary here. A 64-bit counter will never wrap. Even if you increment it every nanosecond, it will not overflow for about 300 years.

Reply:
LOL okay sweet

Reply:
Done.

Reply:
@kevpar are you okay with this? Do you still think we should make a change for the "suggested" uvm path?

Reply:
I think we should try to either do all of the UVM path determination in `hcsoci`, or in `uvm`, but I don't really like the current approach where we do some in each.

@ambarve do you want to move the logic into `hcsoci`?

Reply:
Yea me and John talked about this today, I'd assume this is my next task to make what each package does more uniform rather than the way it is now... all over the place.

Reply:
@kevpar Should I do that in this change or should we wait for @Cantasaurus to take care of it in his PR?

Reply:
We can wait and do it later.